src/condor_utils/print_format_tables.cpp

```diff
@@ -0,0 +1,947 @@
+		{ TRANSFERRING_OUTPUT, "XFER_OUT" },
+	};
+	for (size_t ii = 0; ii < COUNTOF(states); ++ii) {
+		if (jobStatus == states[ii].status) {
+			result = states[ii].psz;
+			return true;
+		}
+	}
+	formatstr(result, "%d", jobStatus);
+	return true;
+}
+
+bool render_hist_runtime (std::string & out, ClassAd * ad, Formatter & /*fmt*/)
+{
+	double utime;
+	if(!ad->LookupFloat(ATTR_JOB_REMOTE_WALL_CLOCK,utime)) {
+		if(!ad->LookupFloat(ATTR_JOB_REMOTE_USER_CPU,utime)) {
+			utime = 0;
+		}
+	}
+	out = format_time((time_t)utime);
+	return (time_t)utime != 0;
+}
+
+bool render_job_cmd_and_args (std::string & val, ClassAd * ad, Formatter & /*fmt*/)
+{
+	if ( ! ad->LookupString(ATTR_JOB_CMD, val))
+		return false;
+
+	char * args;
+	if (ad->LookupString (ATTR_JOB_ARGUMENTS1, &args) || 
+		ad->LookupString (ATTR_JOB_ARGUMENTS2, &args)) {
+		val += " ";
+		val += args;
+		free(args);
+	}
+	return true;
+}
+
+bool render_job_description (std::string & out, ClassAd *ad, Formatter &)
+{
+	if ( ! ad->LookupString(ATTR_JOB_CMD, out))
+		return false;
+
+	std::string description;
+	if ( ! ad->LookupString("MATCH_EXP_" ATTR_JOB_DESCRIPTION, description)) {
+		ad->LookupString(ATTR_JOB_DESCRIPTION, description);
+	}
+	if ( ! description.empty()) {
+		formatstr(out, "(%s)", description.c_str());
+	} else {
+		std::string put_result = condor_basename(out.c_str());
+		std::string args_string;
+		ArgList::GetArgsStringForDisplay(ad,args_string);
+		if ( ! args_string.empty()) {
+			formatstr_cat(put_result, " %s", args_string.c_str());
+		}
+		out = put_result;
+	}
+	return true;
+}
+
+const char * format_job_factory_mode (const classad::Value &val, Formatter &)
+{
+	if (val.IsUndefinedValue()) return "";
+	int pause_mode = 0;
+	if (val.IsNumber(pause_mode)) {
+		switch(pause_mode) {
+		case -1: return "Errs";  // InvalidSubmit
+		case 0:  return "Norm";  // Running
+		case 1:  return "Held";  // Hold
+		case 2:  return "Done";  // Done
+		case 3:  return "Gone";  // Removed
+		default: return "Unk";   // 
+		}
+	} else {
+		return "????";
+	}
+}
+
+bool render_job_id (std::string & result, ClassAd* ad, Formatter &)
+{
+	char str[PROC_ID_STR_BUFLEN];
+	int cluster_id=0, proc_id=0;
+	if ( ! ad->LookupInteger(ATTR_CLUSTER_ID, cluster_id))
+		return false;
+	ad->LookupInteger(ATTR_PROC_ID,proc_id);
+	ProcIdToStr(cluster_id, proc_id, str);
+	result = str;
+	return true;
+}
+
+bool render_job_status_char (std::string & result, ClassAd*ad, Formatter &)
+{
+	int job_status;
+	if ( ! ad->LookupInteger(ATTR_JOB_STATUS, job_status))
+		return false;
+
+	char put_result[3];
+	put_result[1] = ' ';
+	put_result[2] = 0;
+
+	put_result[0] = encode_status(job_status);
+	/* The suspension of a job is a second class citizen and is not a true
+	*  status that can exist as a job status ad and is instead
+	*  inferred, so therefore the processing and display of
+	*  said suspension is also second class.
+	*/
+	if (param_boolean("REAL_TIME_JOB_SUSPEND_UPDATES", false)) {
```

Comment:
UGH.  I did not know that there was a param_boolean call inside the formatting function.  that's bad.